src/mpid/ch4/netmod/ofi/ofi_am_events.h

```diff
@@ -342,7 +342,7 @@ static inline int MPIDI_OFI_handle_lmt_ack(MPIDI_OFI_am_header_t * msg_hdr)
         MPIR_Assert(MPIDI_OFI_KEY_TYPE_HUGE_RMA == key_type);
 
         MPIDI_OFI_index_allocator_free(MPIDI_OFI_COMM(MPIR_Process.comm_world).rma_id_allocator,
-                                       idx_back);
+                                       (uint64_t) idx_back);
```

Comment:
Are we keeping this cast? Why are we not storing the returned key as a `uint64_t`? That is what libfabric defines in the API after all.

Reply:
For this specific patch we do need this cast. However, this will be removed in later commits when key layout is simplified. At the point, the `MPIDI_OFI_rma_key_unpack()` is gone and we will be using 64bit flat key space.